step-capstone/src/components/AddHotel.js

```diff
@@ -0,0 +1,186 @@
+    const [checked, setChecked] = useState(overwriting ? props.data.finalized : false);
+    const [title, setTitle] = useState(overwriting ? props.data.title : "");
+    const [location, setLocation] = useState(overwriting ? props.data.location : "");
+    const [description, setDescription] = useState(overwriting ? props.data.description : "");
+ 
+  
+   const handleChecked = (e) => {
+       setChecked(e.target.checked);
+   }
+ 
+   const handleTitleChange = (e) => {
+       setTitle(e.target.value);
+   }
+ 
+   const handleLocationChange = (e) => {
+       setLocation(e.target.value);
+   }
+ 
+   const handleDescriptionChange = (e) => {
+       setDescription(e.target.value);
+   }
+   
+   const handleStartDateChange = (newStartDate) => { 
+        if (newStartDate >= endDate) {
+            var newEndDate = new Date(newStartDate);
+            newEndDate.setTime(newStartDate.getTime() + 5*60000);
+            setEndDate(newEndDate);
+        }
+        setStartDate(newStartDate);
+    }
+    const handleEndDateChange = (newEndDate) => {
+        if (startDate >= newEndDate) {
+            var newStartDate = new Date(newEndDate);
+            newStartDate.setTime(newEndDate.getTime() - 5*60000);
+            setStartDate(newStartDate);
+        }
+        setEndDate(newEndDate);
+    }
+
+   /*
+    * Called once change to hook state is complete. Updates data property in AddForm.
+    */
+   useEffect(() => {
+ 
+       props.onDataChange({
+           id: overwriting ? props.data.id : undefined,
+           title: title,
+           type: "hotel",
+           startDate: startDate,
+           endDate: endDate,
+           finalized: checked,
+           location: location,
+           description: description
+       })
+ 
+       //validating input
+       if (title === "") {
+           props.onToggleValidation(false);
+       } else if (checked && location === "") {
+           props.onToggleValidation(false);
+       } else {
+           props.onToggleValidation(true);
+       }
+   }, [startDate, endDate, checked, title, location, description])
+ 
+   // render error input if no title
+   let renderTitleInput = () => {
+       if (title === "") {
+           return (
+               <TextField
+                   error
+                   helperText="Cannot leave field blank"
+                   id="title"
+                   label={"Add Title"}
+                   defaultValue={title}
+                   fullWidth
+                   onChange={handleTitleChange}
+               />
+           )
+       } else {
+           return (
+               <TextField
+                   id="title"
+                   label={"Add Title"}
+                   defaultValue={title}
+                   fullWidth
+                   onChange={handleTitleChange}
+               />
+           )
+       }
+   }
+ 
+   // Render error input if finalized item has no location
+   let renderLocationInput = () => {
```

Comment:
Instead of having two cases for two different text fields, you can merge it into one (without this render function) that sets error = null and helperText = "" if no error exists.

Reply:
Good suggest for more concise code! I will make changes!